Jenkinsfile

```diff
@@ -16,7 +10,147 @@ pipeline {
+                            label 'dind'
+                            args '-v /var/lib/docker/.gradle:/root/.gradle'
+                        }
+                    }
+                    steps {
+                        echo env.BRANCH_NAME
+                        dir('uaa-cf-release') {
+                            git changelog: false, credentialsId: 'github.build.ge.com', poll: false, url: 'https://github.build.ge.com/predix/uaa-cf-release.git', branch: 'feature/jenkinsfile'
+                        }
+                        dir('uaa') {
+                            checkout scm
+                        }
+                        sh '''#!/bin/bash -ex
+                                source uaa-cf-release/config-local/set-env.sh
+                                unset HTTPS_PROXY
+                                unset HTTP_PROXY
+                                unset http_proxy
+                                unset https_proxy
+                                unset GRADLE_OPTS
+                                pushd uaa
+                                    ./gradlew --no-daemon --continue :cloudfoundry-identity-server:test
+                                popd
+                                '''
+                    }
+                    post {
+                        success {
+                            echo "Unit tests completed"
+                        }
+                        failure {
+                            echo "Unit tests failed"
+                        }
+                    }
+                }
+                stage('Mockmvc Tests') {
+                    when {
+                        expression { true }
+                    }
+                    agent {
+                        docker {
+                            image 'repo.ci.build.ge.com:8443/predix-security/uaa-ci-testing:0.0.5'
+                            label 'dind'
+                            args '-v /var/lib/docker/.gradle:/root/.gradle'
+                        }
+                    }
+                    steps {
+                        echo env.BRANCH_NAME
+                        dir('uaa-cf-release') {
+                            git changelog: false, credentialsId: 'github.build.ge.com', poll: false, url: 'https://github.build.ge.com/predix/uaa-cf-release.git', branch: 'feature/jenkinsfile'
+                        }
+                        dir('uaa') {
+                            checkout scm
+                        }
+                        sh '''#!/bin/bash -ex
+                            source uaa-cf-release/config-local/set-env.sh
+                            unset HTTPS_PROXY
+                            unset HTTP_PROXY
+                            unset http_proxy
+                            unset https_proxy
+                            unset GRADLE_OPTS
+                            pushd uaa
+                                apt-get -qy install lsof
```

Comment:
should we put this in the docker image instead?

Reply:
Cleaner vs leaner docker image. We can add this to the refactor story